tests/abi/tools/revng-check-compatibility-with-abi/ABIRuntimeTestResultParser.cpp

```diff
@@ -11,25 +11,101 @@
 
 #include "ABIRuntimeTestResultParser.h"
 
-namespace State {
+namespace abi::runtime_test {
+
+struct RawState {
+  std::vector<RawWord> Registers;
+  std::vector<std::byte> Stack;
+
+  bool empty() const { return Registers.empty() && Stack.empty(); }
+
+  State extract(model::Architecture::Values Architecture) && {
+    SortedVector<Register> ResultRegisters;
+    ResultRegisters.reserve(model::Register::getCount(Architecture));
+    for (auto I = ResultRegisters.batch_insert();
+         auto Reg : model::Architecture::registers(Architecture)) {
+      constexpr auto PoN = model::PrimitiveTypeKind::PointerOrNumber;
+      if (model::Register::primitiveKind(Reg) != PoN) {
+        // Only support generic registers for now.
+        // TODO: add vector register support.
+        continue;
+      }
+
+      auto Iterator = llvm::find_if(Registers, [Reg](const RawWord &Raw) {
+        return Raw.Name == getRegisterName(Reg);
+      });
+      if (Iterator == Registers.end()) {
+        std::string Error = "Unable to find '" + getRegisterName(Reg).str()
+                            + "' in the artifact.\n";
+        revng_abort(Error.c_str());
+      }
+      I.emplace(std::move(*Iterator), Architecture);
+    }
 
-using namespace abi::artifact;
+    return State(std::move(ResultRegisters), std::move(Stack));
+  }
+};
 
-struct SingleRun {
+struct SingleIteration {
```

Comment:
"Single" is the opposite of "multiple".
This is just an iteration.

Reply:
Done.